thrift/shared.thrift

```diff
@@ -114,6 +114,12 @@ exception StickyWorkerUnavailableError {
   1: required string message
 }
 
+exception TaskListNotOwnedByHostError {
```

Comment:
I wondered if it would be worth having a `NotOwnedByHostError` for use in both history and matching and while I can see that being slightly less annoying, it may not be great ultimately to tie the two errors together, so I ultimately don't really have a strong opinion on this. It may be easier to evolve them separately.